go/vt/binlog/eventtoken/compare.go

```diff
@@ -32,17 +30,47 @@ func Minimum(ev1, ev2 *querypb.EventToken) *querypb.EventToken {
 	}
 }
 
-// Fresher returns true if ev1 is fresher than ev2. In case of doubt,
-// it returns false.
-//
-// FIXME(alainjobart): if both events come from the same shard, we can
-// compare replication position (after parsing it). Need to implement that.
-// FIMXE(alainjobart): when it's more than just a simple compare, add
-// unit tests too.
-func Fresher(ev1, ev2 *querypb.EventToken) bool {
+// Fresher compares two event tokens.  It returns a negative number if
+// ev1<ev2, zero if they're equal, and a positive number if
+// ev1>ev2. In case of doubt (we don't have enough information to know
+// for sure), it returns -1.
```

Comment:
This last sentence makes it sound like it's possible to distinguish the "doubt" state from others, but actually it's indistinguishable from `ev1<ev2`. I would just say it returns "a negative number" like above, so it doesn't appear to make promises.